roles/tls/tasks/main.yml

```diff
@@ -0,0 +1,87 @@
+# code: language=ansible
+---
+- name: Ensure directories exist
+  ansible.builtin.file:
+    path: "{{ generated_certs_dir }}"
+    state: directory
+    mode: '0755'
+    recurse: true
+
+- name: Generate CA private key
+  openssl_privatekey:
+    path: "{{ ca_key_path }}"
+    size: 4096
+    mode: "0644"
+  register: ca_key
+
+- name: Create CA certificate signing request
+  community.crypto.openssl_csr:
+    path: "/tmp/ca.csr"
```

Comment:
maybe we should use `generated_certs_dir` instead of hardcoding `/tmp/ca.csr`.
what do you think?
the same for all keys going directly to `/tmp` folder

Reply:
Done in https://github.com/trento-project/ansible/pull/57/commits/aa1011b30c6924100af7d70fe5267b34b782c71c